src/systems/fem_context.C

```diff
@@ -1305,18 +1305,20 @@ void FEMContext::elem_reinit(Real theta)
 
   // Handle a moving element if necessary.
   if (_mesh_sys)
-    // We assume that the ``default'' state
-    // of the mesh is its final, theta=1.0
-    // position, so we don't bother with
-    // mesh motion in that case.
-    if (theta != 1.0)
-      {
-        // FIXME - ALE is not threadsafe yet!
-        libmesh_assert_equal_to (libMesh::n_threads(), 1);
+    {
+      // We assume that the ``default'' state
+      // of the mesh is its final, theta=1.0
+      // position, so we don't bother with
+      // mesh motion in that case.
+      if (theta != 1.0)
+        {
+          // FIXME - ALE is not threadsafe yet!
+          libmesh_assert_equal_to (libMesh::n_threads(), 1);
 
-        elem_position_set(theta);
-      }
-  elem_fe_reinit();
+          elem_position_set(theta);
+        }
+      elem_fe_reinit();
+   }
 }
```

Comment:
> Nah, I mean making it clearer that we're not reinitializing FE objects unless a moving mesh requires it.

Ah,OK.

> Originally elem_reinit was supposed to reinitialize FE objects; and if you look at the documentation in DiffSystem that's clearly implied. It wasn't until later optimization that I broke out elem_fe_reinit() as a separate method.

Thanks for the clarification. I don't remember a time elem_fe_reinit wasn't called in the AssemblyContributions functor. But my memory has never been the same since my kids were born.